src/block-management/block-holder.js

```diff
@@ -45,36 +54,109 @@ export class BlockHolder extends React.Component<PropsType, StateType> {
 		};
 	}
 
-	renderToolbarIfBlockFocused() {
-		if ( this.props.isSelected ) {
-			return (
-				<InlineToolbar
-					clientId={ this.props.clientId }
-					onButtonPressed={ this.props.onInlineToolbarButtonPressed }
-					canMoveUp={ this.props.canMoveUp }
-					canMoveDown={ this.props.canMoveDown }
-				/>
-			);
+	onFocus = ( event ) => {
+		if ( event ) {
+			// == Hack for the Alpha ==
+			// When moving the focus from a TextInput field to another kind of field the call that hides the keyboard is not invoked
+			// properly, resulting in keyboard up when it should not be there.
+			// The code below dismisses the keyboard (calling blur on the last TextInput field) when the field that now gets the focus is a non-textual field
+			const currentlyFocusedTextInput = TextInputState.currentlyFocusedField();
+			if ( event.nativeEvent.target !== currentlyFocusedTextInput && ! TextInputState.isTextInput( event.nativeEvent.target ) ) {
+				TextInputState.blurTextInput( currentlyFocusedTextInput );
+			}
 		}
+		this.props.onSelect( this.props.clientId );
+	};
+
+	onInlineToolbarButtonPressed = ( button: number, clientId: string ) => {
+		switch ( button ) {
+			case InlineToolbarActions.UP:
+				this.props.moveBlocksUp( clientId );
```

Comment:
I'm a bit bugged that we're accepting and passing `clientId` here even though the clientId is a constant for each block. Is it perhaps an anti-pattern to modify the `withDispatch` to return a curried version of `moveBlocksUp` that doesn't require the parameter? Same for numerous other functions passed as props from `withDispatch` that require `clientId` but don't exactly need it locally. WDYT?

Reply:
Agreed 👍

Reply:
Can we do the same for `getPreviousBlockClientId()` and `getNextBlockClientId()`?

Reply:
> Can we do the same for getPreviousBlockClientId() and getNextBlockClientId()?

Answering my own question, those fall under the same reasoning as https://github.com/wordpress-mobile/gutenberg-mobile/pull/335#discussion_r245754229 so, leaving them as is 👍.